test/FieldMatrix.jl

```diff
@@ -0,0 +1,104 @@
+@testset "FieldMatrix" begin
+    @testset "Immutable Tensor3x3" begin
+        eval(quote
+            struct Tensor3x3 <: FieldMatrix{3, 3, Float64}
+                xx::Float64
+                xy::Float64
+                xz::Float64
+                yx::Float64
+                yy::Float64
+                yz::Float64
+                zx::Float64
+                zy::Float64
+                zz::Float64
+            end
+
+            StaticArrays.similar_type(::Type{Tensor3x3}, ::Type{Float64}, s::Size{(3,3)}) = Tensor3x3
+        end)
+
+        p = Tensor3x3(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0)
+
+        @test_throws Exception p[2] = 4.0
+        @test_throws Exception p[2, 3] = 6.0
+
+        @test size(p) === (3,3)
+        @test length(p) === 9
+        @test eltype(p) === Float64
+
+        @testinf Tuple(p) === (1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0)
+
+        @test (p + p) === Tensor3x3(2.0, 4.0, 6.0, 8.0, 10.0, 12.0, 14.0, 16.0, 18.0)
```

Comment:
This could do with an `@inferred` on the left hand side

Reply:
Sure thing!